### PR TITLE
Increase test coverage

### LINKED ISSUE #463: OpenSSF Best Practices "gold" badge
OpenSSF Best Practices "gold" badge:

**Change Control**:

- [x]  The project MUST clearly identify small tasks that can be performed by new or casual contributors.
- [x] The project MUST require two-factor authentication (2FA) for developers for changing a central repository or accessing sensitive data (such as private vulnerability reports). This 2FA mechanism MAY use mechanisms without cryptographic mechanisms such as SMS, though that is not recommended. 
- [x] The project's two-factor authentication (2FA) SHOULD use cryptographic mechanisms to prevent impersonation. Short Message Service (SMS) based 2FA, by itself, does NOT meet this criterion, since it is not encrypted.

**Quality**

- [x] The project MUST document its code review requirements, including how code review is conducted, what must be checked, and what is required to be acceptable.
- [x] The project MUST have at least 50% of all proposed modifications reviewed before release by a person other than the author, to determine if it is a worthwhile modification and free of known issues which would argue against its inclusion 
- [x] The project MUST have a reproducible build. If no building occurs (e.g., scripting languages where the source code is used directly instead of being compiled), select "not applicable" (N/A)
- [x] The project MUST implement continuous integration, where new or changed code is frequently integrated into a central code repository and automated tests are run on the result
- [x] The project MUST have FLOSS automated test suite(s) that provide at least 90% statement coverage if there is at least one FLOSS tool that can measure this criterion in the selected language.
- [x] The project MUST have FLOSS automated test suite(s) that provide at least 80% branch coverage if there is at least one FLOSS tool that can measure this criterion in the selected language.

**Security** 
- [x] The project website, repository (if accessible via the web), and download site (if separate) MUST include key hardening headers with nonpermissive values.
- [x] The project MUST have performed a security review within the last 5 years. This review MUST consider the security requirements and security boundary. 
- [x] Hardening mechanisms MUST be used in the software produced by the project so that software defects are less likely to result in security vulnerabilities.

### PR DESCRIPTION
**Pull request by tdrozdovsky**
Signed-off-by: Taras Drozdovskyi <t.drozdovsky@samsung.com>

# Description
Achieving a test coverage of more than 80% will get a OpenSSF Best Practices "gold" badge

Fixes #463 

## Type of change

- [x] Test Coverage update

# How Has This Been Tested?
```
make test
```

```
github.com/lf-edge/edge-home-orchestration-go/internal/common/commandvalidator	96.77%	30/31
github.com/lf-edge/edge-home-orchestration-go/internal/common/commandvalidator/blacklist	100.00%	1/1
github.com/lf-edge/edge-home-orchestration-go/internal/common/commandvalidator/commands	100.00%	12/12
github.com/lf-edge/edge-home-orchestration-go/internal/common/commandvalidator/injectionchecker	100.00%	4/4
github.com/lf-edge/edge-home-orchestration-go/internal/common/errormsg	100.00%	10/10
github.com/lf-edge/edge-home-orchestration-go/internal/common/errors	100.00%	8/8
github.com/lf-edge/edge-home-orchestration-go/internal/common/fscreator	100.00%	5/5
github.com/lf-edge/edge-home-orchestration-go/internal/common/logmgr	100.00%	28/28
github.com/lf-edge/edge-home-orchestration-go/internal/common/mqtt	96.90%	125/129
github.com/lf-edge/edge-home-orchestration-go/internal/common/networkhelper	81.44%	79/97
github.com/lf-edge/edge-home-orchestration-go/internal/common/networkhelper/detector	95.00%	19/20
github.com/lf-edge/edge-home-orchestration-go/internal/common/requestervalidator	100.00%	8/8
github.com/lf-edge/edge-home-orchestration-go/internal/common/requestervalidator/requesterstore	100.00%	12/12
github.com/lf-edge/edge-home-orchestration-go/internal/common/resourceutil	57.94%	124/214
github.com/lf-edge/edge-home-orchestration-go/internal/common/resourceutil/cpu	98.11%	104/106
github.com/lf-edge/edge-home-orchestration-go/internal/common/sigmgr	83.33%	10/12
github.com/lf-edge/edge-home-orchestration-go/internal/controller/configuremgr	79.78%	71/89
github.com/lf-edge/edge-home-orchestration-go/internal/controller/discoverymgr	82.37%	341/414
github.com/lf-edge/edge-home-orchestration-go/internal/controller/discoverymgr/mnedc	78.79%	78/99
github.com/lf-edge/edge-home-orchestration-go/internal/controller/discoverymgr/mnedc/client	73.09%	163/223
github.com/lf-edge/edge-home-orchestration-go/internal/controller/discoverymgr/mnedc/server	88.83%	167/188
github.com/lf-edge/edge-home-orchestration-go/internal/controller/scoringmgr	100.00%	65/65
github.com/lf-edge/edge-home-orchestration-go/internal/controller/securemgr/authenticator	48.21%	27/56
github.com/lf-edge/edge-home-orchestration-go/internal/controller/securemgr/authorizer	92.31%	36/39
github.com/lf-edge/edge-home-orchestration-go/internal/controller/securemgr/verifier	93.46%	100/107
github.com/lf-edge/edge-home-orchestration-go/internal/controller/servicemgr	90.00%	63/70
github.com/lf-edge/edge-home-orchestration-go/internal/controller/servicemgr/executor	100.00%	2/2
github.com/lf-edge/edge-home-orchestration-go/internal/controller/servicemgr/executor/androidexecutor	68.00%	34/50
github.com/lf-edge/edge-home-orchestration-go/internal/controller/servicemgr/executor/containerexecutor	86.25%	414/480
github.com/lf-edge/edge-home-orchestration-go/internal/controller/servicemgr/executor/nativeexecutor	97.50%	39/40
github.com/lf-edge/edge-home-orchestration-go/internal/controller/servicemgr/notification	92.50%	37/40
github.com/lf-edge/edge-home-orchestration-go/internal/controller/storagemgr/config	100.00%	17/17
github.com/lf-edge/edge-home-orchestration-go/internal/controller/storagemgr/storagedriver	43.88%	129/294
github.com/lf-edge/edge-home-orchestration-go/internal/db/bolt/application	86.11%	31/36
github.com/lf-edge/edge-home-orchestration-go/internal/db/bolt/common	100.00%	6/6
github.com/lf-edge/edge-home-orchestration-go/internal/db/bolt/configuration	83.67%	41/49
github.com/lf-edge/edge-home-orchestration-go/internal/db/bolt/network	74.58%	44/59
github.com/lf-edge/edge-home-orchestration-go/internal/db/bolt/resource	84.62%	22/26
github.com/lf-edge/edge-home-orchestration-go/internal/db/bolt/service	84.00%	42/50
github.com/lf-edge/edge-home-orchestration-go/internal/db/bolt/system	84.62%	22/26
github.com/lf-edge/edge-home-orchestration-go/internal/db/bolt/wrapper	84.13%	53/63
github.com/lf-edge/edge-home-orchestration-go/internal/db/helper	100.00%	45/45
github.com/lf-edge/edge-home-orchestration-go/internal/orchestrationapi	59.39%	155/261
github.com/lf-edge/edge-home-orchestration-go/internal/restinterface	100.00%	1/1
github.com/lf-edge/edge-home-orchestration-go/internal/restinterface/cipher	100.00%	2/2
github.com/lf-edge/edge-home-orchestration-go/internal/restinterface/cipher/dummy	94.74%	18/19
github.com/lf-edge/edge-home-orchestration-go/internal/restinterface/cipher/sha256	88.33%	53/60
github.com/lf-edge/edge-home-orchestration-go/internal/restinterface/client	0.00%	0/1
github.com/lf-edge/edge-home-orchestration-go/internal/restinterface/client/restclient	100.00%	122/122
github.com/lf-edge/edge-home-orchestration-go/internal/restinterface/externalhandler	78.73%	248/315
github.com/lf-edge/edge-home-orchestration-go/internal/restinterface/externalhandler/senderresolver	98.00%	49/50
github.com/lf-edge/edge-home-orchestration-go/internal/restinterface/internalhandler	96.79%	181/187
github.com/lf-edge/edge-home-orchestration-go/internal/restinterface/resthelper	84.21%	64/76
github.com/lf-edge/edge-home-orchestration-go/internal/restinterface/resthelper/client/httphelper	50.00%	1/2
github.com/lf-edge/edge-home-orchestration-go/internal/restinterface/resthelper/client/tlshelper	62.50%	20/32
github.com/lf-edge/edge-home-orchestration-go/internal/restinterface/route	89.19%	33/37
github.com/lf-edge/edge-home-orchestration-go/internal/restinterface/route/tlsserver	100.00%	20/20
github.com/lf-edge/edge-home-orchestration-go/internal/restinterface/tls	100.00%	11/11
Report Total	80.56%	3646/4526
```
**Test Configuration**:
* OS type & version: Ubuntu 20.04 
* Hardware: x86-64 
* Toolchain: Docker v20.10 & Go v1.16
* Edge Orchestration Release: v1.1.x

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: internal/common/resourceutil/cpu/cpu.go [hunk 1]
```diff
@@ -182,7 +183,8 @@ func getCPUFreqCPUInfo() (float64, error) {
 
 	r := bufio.NewReader(f)
 	for {
-		line, err := r.ReadString('\n')
+		var line string
+		line, err = r.ReadString('\n')
```
Comment (t25kim) [inline internal/common/resourceutil/cpu/cpu.go:187]: I don't know why the 'line' variable should be declared like this. Is there a problem with the previous code?
If necessary, what about declaring `line` in front of `for {`?

Comment (tdrozdovsky) [reply internal/common/resourceutil/cpu/cpu.go]: In the previous code, when an error occurred in the `for { ... }`, the `err` variable did not return an error when exiting the function

Comment (tdrozdovsky) [reply internal/common/resourceutil/cpu/cpu.go]: > If necessary, what about declaring `line` in front of `for {`?

Done! Thank you for advice!

File: internal/common/resourceutil/cpu/cpu.go [hunk 2]
```diff
@@ -181,8 +182,9 @@ func getCPUFreqCPUInfo() (float64, error) {
 	defer f.Close()
 
 	r := bufio.NewReader(f)
+	var line string
 	for {
-		line, err := r.ReadString('\n')
+		line, err = r.ReadString('\n')
 		if err != nil {
 			break
```
Comment (suresh-lc) [inline internal/common/resourceutil/cpu/cpu.go:189]: Should we add a error log before break

Comment (tdrozdovsky) [reply internal/common/resourceutil/cpu/cpu.go:190]: Done!

**Review by t25kim — APPROVED** [review]
LGTM Thanks!!

**Review by MoonkiHong — APPROVED** [review]
+1

**Review by suresh-lc — APPROVED** [review]
+1

**Pull request merged**